src/commands/image/imageSource/buildImageInAzure/RunStep.ts

```diff
@@ -28,10 +30,18 @@ export class RunStep extends ExecuteActivityOutputStepBase<BuildImageInAzureImag
                 dockerFilePath: path.basename(context.dockerfilePath) /* Assume the dockerfile is always in the root of the source */
             };
 
-            const building: string = localize('buildingImage', 'Building image...');
-            progress.report({ message: building });
-
-            context.run = await context.client.registries.beginScheduleRunAndWait(context.resourceGroupName, context.registryName, runRequest);
+            const retries = 3;
+            await retry(
+                async (currentAttempt: number) => {
+                    context.run = await context.client.registries.beginScheduleRunAndWait(context.resourceGroupName, context.registryName, runRequest);
```

Comment:
I think repositioning the messages up will likely allow you to avoid adding the `onFailedAttempt` callback after all.  That along with the findings in the comments below :)